### PR TITLE
Graphical representation of clusters

### LINKED ISSUE #773: Adding visual represnetation of matrices
It would be nice to have a visual representation of the clusters obtained from the modules `clustfcc` and `clustrmsd`.

### PR DESCRIPTION
**Pull request by VGPReys**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [X] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [ ] Your PR is about CNS
- [X] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [X] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [X] Your code follows our coding style
- [x] You wrote tests for the new code
- [X] `tox` tests pass. *Run `tox` command inside the repository folder*
- [X] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [X] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [X] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

Adding the possibility to generate a graphical representation of the `clustfcc` or `clustrmsd` matrices.
A new parameters was added in both modules: `plot_matrix` = true / false, enabling to generate a plotly representation of the matrices.
The corresponding heatmap(s) is/are ordered by cluster size.

Closes #773

### CODE REVIEW

File: src/haddock/modules/analysis/clustrmsd/clustrmsd.py [hunk 1]
```diff
@@ -48,7 +45,32 @@ def read_matrix(rmsd_matrix: RMSDFile) -> np.ndarray:
             else:
                 matrix[[c]] = float(data[2])
                 c += 1
-    return matrix
+    return np.array(matrix)
+
+
+def get_matrix_path(rmsd_matrix: RMSDFile) -> Path:
```
Comment (mgiulini) [inline src/haddock/modules/analysis/clustrmsd/clustrmsd.py:51]: shouldn't this go above the previous function?

File: src/haddock/modules/analysis/clustfcc/__init__.py [hunk 2]
```diff
@@ -250,4 +263,25 @@ def _run(self) -> None:
             log.warning("No clusters were found")
             self.output_models = models_to_cluster  # type: ignore
 
+        # Draw the matrix
+        if self.params['plot_matrix']:
+            # Obtain final models indices
+            final_order_idx, labels = [], []
+            for pdb in self.output_models:
+                final_order_idx.append(models_to_cluster.index(pdb))
```
Comment (mgiulini) [inline src/haddock/modules/analysis/clustfcc/__init__.py:271]: If I understand correctly this means that there's no re-ordering of models prior to plotting, right?

File: src/haddock/libs/libclust.py [hunk 3]
```diff
@@ -48,3 +54,78 @@ def write_structure_list(input_models: list[PDBFile],
...
+    """Plot a plotly heatmap of a matrix file.
+
+    Parameters
+    ----------
+    matrix_path : Union[Path, FilePath, str]
+        Path to a half-matrix
+    final_order_idx : list[int]
+        Index orders
+    labels : list[str]
+        Ordered labels
```
Comment (mgiulini) [inline src/haddock/libs/libclust.py:78]: I would add the angstrom unit

**Review by mgiulini — COMMENTED** [review]
the plots look nice and informative! I added a few minor comments. I have only one major concern: should we limit by default the maximum number of entries for plotting? for example to 2000 or something like that? Plotting a bigger matrix can become heavy on disk and quite uninformative..

File: src/haddock/modules/analysis/clustfcc/__init__.py (hunk 2 above)
Comment (VGPReys) [reply src/haddock/modules/analysis/clustfcc/__init__.py:271]: Yes, the ordering is taken from the self.output_models, for agreements with next modules.
At this stage, they are ordered by cluster sizes, as found in `cluster.out` file minus the ones excluded from the parameters.

**Review by mgiulini — APPROVED** [review]

**Pull request merged**